src/jshint.js

```diff
@@ -4247,12 +4305,12 @@ var JSHINT = (function() {
     x.lbp = 25;
   }(prefix("yield", function() {
     var prev = state.tokens.prev;
-    if (state.inESNext(true) && !state.funct["(generator)"]) {
+    if (state.inES6(true) && !state.funct["(generator)"]) {
```

Comment:
Sorry for the confusion, generators did make es6, this thread only applies to arrray comprehensions.